skt/executable.py

```diff
@@ -443,11 +445,19 @@ def cmd_run(cfg):
     run_report_path = join_with_slash(cfg.get('output_dir'), 'run.report')
 
     runner = skt.runner.getrunner(*cfg.get('runner'))
-    retcode, report_string = runner.run(cfg.get('buildurl'),
-                                        cfg.get('max_aborted_count'),
-                                        cfg.get('krelease'),
-                                        cfg.get('wait'),
-                                        arch=cfg.get("kernel_arch"))
+
+    atexit.register(runner.cleanup_handler)
+    signal.signal(signal.SIGINT, runner.signal_handler)
```

Comment:
This would only hit with `SIGINT` but not with `SIGTERM`, `SIGKILL` or any other signals. Not sure what GitLab does to kill skt (maybe @agustinhenze can help here?) but it would be better to use something more generic here than `^C`

Reply:
We can add SIGTERM, but SIGKILL by its very nature cannot be trapped. 

http://en.wikipedia.org/wiki/Unix_signal#SIGKILL

Reply:
Right, forgot about that detail for a while. Real fun will be if that's what GitLab/UpShift use :D

Reply:
From what I've seen so far, we have no way of handling termination on GitLab/OpenShift, we're likely getting a SIGKILL there. At least GitLab has an issue asking to switch to a softer shutdown: SIGTERM with timeout to SIGKILL.

Reply:
I don't see much else we can do from this end. So how about merging this issue, leaving FM-357 open or creating a follow-up ticket and asking Don to put a bit of pressure on that GitLab/OpenShift issue?

Reply:
I would say that skt should still handle SIGTERM in addition to SIGINT. After that, and provided @veruu is OK with all the rest, I think we can merge it and close FASTMOVING-357. We need to open a separate issue regarding our GitLab pipeline not cleaning up, and mention the [upstream issue](https://gitlab.com/gitlab-org/gitlab-runner/issues/3031) there.

We cannot really put a pressure on GitLab, but we can suggest an implementation, or at least give the issue a thumbs-up in hope that it would raise the priority. @agustinhenze, is there anything else we can do there?

Reply:
Look underneath, V2 has SIGTERM and SIGINT.

Reply:
Ah, great, thank you, Jakub :)